### DISCUSSION TITLE
How does one insert a literal newline (or carriage return) in a mapping?

### DISCUSSION THREAD

**Discussion by huyz** [esm7/obsidian-vimrc-support#239] [Q&A] [answered]
These mappings used to work for me:
```
exmap blankBelow obcommand obsidian-editor-shortcuts:insertLineBelow
exmap blankAbove obcommand obsidian-editor-shortcuts:insertLineAbove
nmap &a& :blankAbove
nmap &b& :blankBelow
nmap o &b&i
nmap O &a&i
```

That's because there would be an implicit carriage return after `:blankAbove` and `:blankBelow`.
But something changed in the plugin recently that now hitting `o` spits out something like `:blankBelowi`; no more carriage returns.

Does anyone know how I can enter a carriage return, as it's possible in standard .vimrc (`<C-r>`)?

**Comment by esm7** [comment, marked as answer]
Add `<CR>` at the end of each `nmap`. I'll update the documentation once the new Obsidian version is released to all users.

**Comment by dfreder1** [comment]
<CR> works provided there is no space between before it and the nmap command (at least on windows):
nmap J :movedown<CR>

This tripped me up for a while
